test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+'use strict';
+
+var chai = require('chai');
+var expect = chai.expect;
+var chance = require('chance');
+var _ = require('lodash');
+var datasets = require('../../app/datasets');
+
+describe('Lodash Collection training', function () {
+    describe('countBy', function () {
+        it('should count repeating numbers', function () {
+            expect(_.countBy.apply(_, datasets.countBy())).to.eql({1: 1, 2: 2, 6: 2, 7: 3});
+        });
+        it('should count length of given strings', function () {
+            expect(_.countBy.apply(_, datasets.countBy2())).to.eql({3: 2, 4: 3, 5: 1});
+        });
+    });
+
+    describe('every', function () {
+        it('should return true if all elements pass the predicate "ALIVE" check', function () {
+            var params = datasets.every();
+            var array = params[0];
+            expect(array.length).to.eql(2);
+            array.forEach(function (n) {
+                expect(_.has(n, 'ALIVE')).to.eql(true);
+            });
+            expect(_.every.apply(_, datasets.every())).to.eql(true);
+        });
+        it('should return true if all elements are Booleanly true', function () {
+            var params = datasets.every2();
+            var array = params[0];
+            expect(array.length).to.eql(4);
+            expect(_.every.apply(_, datasets.every2())).to.eql(true);
+        });
+    });
+
+    describe('filter', function () {
```

Comment:
filter: once again I've managed to hack tests by just simply specyfing predicate _.property for age and not setting them for the rest of objects. Tests must be much more detailed and specific to ensure that we allow only correct answers according to scenario specifics. Apart from than we need to test other collection/predicate datatypes and I think we should provide(hardcode) array of examples inside datasets and check if user is using specifically them inside tests.